### PR TITLE
Remove hard dependencies

### PR DESCRIPTION
**Pull request by h-vetinari**
See [this](https://github.com/conda-forge/networkx-feedstock/issues/1#issuecomment-1047181328) comment.

Also update python lower bound.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,21 +9,16 @@ source:
...
-    - python >=3.6
+    - python >=3.7
     - pip
-
   run:
-    - python >=3.6
-    - numpy >=1.19
-    - scipy >=1.5,!=1.6.1
-    - matplotlib-base >=3.3
-    - pandas >=1.1
```
Comment (ocefpaf) [inline recipe/meta.yaml:26]: Folks expecting these extras may be surprised with this change. Maybe we should to do a multiple outputs with a "core" package and another one with the extras?

**Comment by MridulS** [comment]
IMO we should remove these hard dependencies too. Currently `pip install networkx` and `conda install -c conda-forge networkx` do not setup the same environment.

This also doesn't match the `defaults` channel https://github.com/AnacondaRecipes/networkx-feedstock/blob/master/recipe/meta.yaml#L22 

We should make sure all these ways of installing networkx setup the same environment:
```
pip install networkx
conda install networkx
conda install -c conda-forge networkx
```

**Comment by ocefpaf** [comment]
@h-vetinari and @MridulS do you want to revive this PR?

**Comment by MridulS** [comment]
I'm +1 on that! @h-vetinari could you rebase this on the `main` branch and remove all the hard dependencies?

**Comment by MridulS** [comment]
moved the changes to #33, closing this :)

**Pull request closed without merging**